### DISCUSSION TITLE
iOS 使用播放多个pag文件，CPU使用率很高达到

### DISCUSSION THREAD

**Discussion by samllOranges** [Tencent/libpag#2392] [Q&A] [answered]
我们是用pag做头饰动画，页面大概有15个控件，随机使用压缩包里面的pag动画文件，CPU使用率多了90%-100%左右。
以下是控件创建使用代码，大概有15个控件这样
PAGView *pageView =[[PAGView alloc] initWithFrame:frame];
        //读取PAG素材文件
        PAGFile *pagFile = [PAGFile Load:path];
        //关联PAGView和PAGFile
        [pageView setComposition:pagFile];
        
        [pageView setPath:path];
   
        [pageView setRepeatCount:0];
        //播放
        [pageView play];

以下是使用的资源文件
[头像框pag测试.zip](https://github.com/user-attachments/files/16427766/pag.zip)

**Comment by domchen** [comment, marked as answer]
素材问题，你们用了一种掩耳盗铃的性能差到极致的素材制作方法：用了无数个图片图层一帧帧轮流播放来避免用BMP预合成。这样只是会在面板上显示看起来没有视频内容，但是实际性能是最差的。除了骗自己没有别的作用。还不如直接整个按BMP预合成导出。你直接重新按BMP预合成导出，然后把PAGView改成用PAGImageView，性能就会飞跃了。以后不要在用这种骗自己的做法了。
